url85940d3d8e_ccp999_Expr4j_tgz-pJ8-org_boris_expr_util_ExprEvaluatorJ8/src/org/boris/expr/engine/Range.java

```diff
@@ -0,0 +1,203 @@
+import org.boris.expr.util.Reflect;
+
+public class Range
+{
+    private String namespace;
+    private String dim1Name;
+    private GridReference dim1;
+    private String dim2Name;
+    private GridReference dim2;
+
+    public Range(String namespace, GridReference dim1) {
+        this.namespace = namespace;
+        this.dim1 = dim1;
+    }
+
+    public Range(String namespace, GridReference dim1, GridReference dim2) {
+        this.namespace = namespace;
+        this.dim1 = dim1;
+        this.dim2 = dim2;
+    }
+
+    public Range(String namespace, String dim1Name, GridReference dim1,
+            String dim2Name, GridReference dim2) {
+        this.namespace = namespace;
+        this.dim1Name = dim1Name;
+        this.dim1 = dim1;
+        this.dim2Name = dim2Name;
+        this.dim2 = dim2;
+    }
+
+    public boolean isArray() {
+        return dim2 != null;
+    }
+
+    public void setNamespace(String namespace) {
+        this.namespace = namespace;
+    }
+
+    public String getNamespace() {
+        return namespace;
+    }
+
+    public String getDimension1Name() {
+        return dim1Name;
+    }
+
+    public GridReference getDimension1() {
+        return dim1;
+    }
+
+    public void setDimension1(GridReference dim1) {
+        this.dim1 = dim1;
+    }
+
+    public String getDimension2Name() {
+        return dim2Name;
+    }
+
+    public GridReference getDimension2() {
+        return dim2;
+    }
+
+    public void setDimension2(GridReference dim2) {
+        this.dim2 = dim2;
+    }
+
+    public int hashCode() {
+        int hc = 0;
+        hc = namespace == null ? 56 : namespace.hashCode();
+        if (dim1 != null)
+            hc ^= dim1.hashCode();
+        else if (dim1Name != null)
+            hc ^= dim1Name.hashCode();
+        if (dim2 != null)
+            hc ^= dim2.hashCode();
+        else if (dim2Name != null)
+            hc ^= dim2Name.hashCode();
+        return hc;
+    }
+
+    public boolean equals(Object obj) {
+        if (!(obj instanceof Range))
+            return false;
+        Range r = (Range) obj;
+        if (!Reflect.equals(r.namespace, namespace))
+            return false;
+        if (!Reflect.equals(r.dim1, dim1))
+            return false;
+        if (!Reflect.equals(r.dim2, dim2))
+            return false;
+        return true;
+    }
+
+    public String toString() {
+        StringBuilder sb = new StringBuilder();
+        if (namespace != null) {
+            sb.append(namespace);
+            sb.append('!');
+        }
+
+        if (dim1Name != null) {
+            sb.append(dim1Name);
+        } else {
+            sb.append(dim1);
+        }
+
+        if (dim2Name != null) {
+            sb.append(':');
+            sb.append(dim2Name);
+        } else if (dim2 != null) {
+            sb.append(':');
+            sb.append(dim2);
+        }
+
+        return sb.toString();
+    }
+
+    public static Range valueOf(String var) throws ExprException {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 12. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.